src/views/self/view_self.cpp

```diff
@@ -184,50 +194,69 @@ namespace big
 
 		ImGui::BeginGroup();
 
-		ImGui::Checkbox("HIDE_RADAR"_T.data(), &g.self.hide_radar);
+		ImGui::Checkbox("HIDE_RADAR"_T.data(), &g.self.hud.hide_radar);
 
 		ImGui::SameLine();
 
-		ImGui::Checkbox("HIDE_AMMO"_T.data(), &g.self.hide_ammo);
+		ImGui::Checkbox("HIDE_AMMO"_T.data(), &g.self.hud.hide_ammo);
 
 		ImGui::SameLine();
 
-		ImGui::Checkbox("FORCE_SHOW_HUD"_T.data(), &g.self.force_show_hud);
+		ImGui::Checkbox("FORCE_SHOW_HUD"_T.data(), &g.self.hud.force_show_hud);
 
-		ImGui::Combo("##hud_comp_combo", &g.self.selected_hud_component, hud_component_names, (int)HudComponents::HUD_WEAPONS);
+		ImGui::Combo("##hud_comp_combo", &g.self.hud.selected_hud_component, hud_component_names, (int)HudComponents::HUD_WEAPONS);
 		ImGui::SameLine();
 		components::button("HIDE"_T, [] {
-			g.self.hud_components_states[g.self.selected_hud_component] = true;
+			g.self.hud.hud_components_states[g.self.hud.selected_hud_component] = true;
 		});
 		ImGui::SameLine();
 		components::button("SHOW"_T, [] {
-			g.self.hud_components_states[g.self.selected_hud_component] = false;
+			g.self.hud.hud_components_states[g.self.hud.selected_hud_component] = false;
 		});
 
 		components::button("HIDE_ALL"_T, [] {
-			g.self.hide_radar = true;
-			g.self.hide_ammo  = true;
+
+			g.self.hud.hide_radar = true;
+			g.self.hud.hide_ammo = true;
+
 			for (int i = 0; i < (int)HudComponents::HUD_WEAPONS; i++)
 			{
-				g.self.hud_components_states[i] = true;
+				g.self.hud.hud_components_states[i] = true;
 			}
 		});
 		ImGui::SameLine();
 		components::button("SHOW_ALL"_T, [] {
-			g.self.hide_radar = false;
-			g.self.hide_ammo  = false;
+
+			g.self.hud.hide_radar = false;
+			g.self.hud.hide_ammo = false;
+
 			for (int i = 0; i < (int)HudComponents::HUD_WEAPONS; i++)
 			{
-				g.self.hud_components_states[i] = false;
+				g.self.hud.hud_components_states[i] = false;
 			}
 		});
 		ImGui::SameLine();
-		ImGui::Checkbox("FORCE_SHOW_HUD_ELEMENT"_T.data(), &g.self.force_show_hud_element);
+		ImGui::Checkbox("FORCE_SHOW_HUD_ELEMENT"_T.data(), &g.self.hud.force_show_hud_element);
 		if (ImGui::IsItemHovered())
 			ImGui::SetTooltip("FORCE_SHOW_HUD_ELEMENT_DESC"_T.data());
 
 		ImGui::EndGroup();
 
+		components::command_checkbox<"hudcolor">();
+
+		ImGui::Checkbox("Ovveride Hud Color IRGBA", &g.self.hud.shcolor);
+		ImGui::InputInt("Hud Index", &g.self.hud.index);//need to display current val if not displayed
```

Comment:
We can just make a map of the HUD Indexes or store them somehow, the average user is just going to change the color and won't understand the index. And seeing as the first is pure white it can get annoying. Also, you spelled Override wrong.

Reply:
i dont know what could really be done to reviuse it, im not that good with imgui